### PR TITLE
Add settings page

### PR DESCRIPTION
**Pull request by nelsonmestevao**

### CODE REVIEW

**Comment by nelsonmestevao** [comment]
#### 📝Changed routes:


| [**/settings**](https://shuriken-lv444mune-cesiuminho.vercel.app/settings) |
|:-:|
|<a href="https://shuriken-lv444mune-cesiuminho.vercel.app/settings">Screenshot of /settings</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/qeroDO5hupppC6DtqFrkSbkuqNNF6dM7FnIDiQxUipx4lDGaokVPstn7k0RA7YSGprNqD7kk2ZMxQdp_aJXhtw.png">(view full size)</a> |







Commit 9f00d8b2a0b06b512345efcb4f7bea1d1a9d50f1 (https://shuriken-lv444mune-cesiuminho.vercel.app).

File: pages/settings.js
Comment (miguelbrandao) [inline pages/settings.js:97]: This label should be 'Nome'

File: pages/settings.js
Comment (miguelbrandao) [inline pages/settings.js:106]: I don't think you should be able to change your name, since kids will be using this and they shouldn't change that data. Actually, there are very few reasons for a name change after a user is registered

**Review by miguelbrandao — COMMENTED** [review]
I'm getting some weird behavior in the previews: when I change some info and save it, the request seems to be successful but it doesn't change it in the view until I do some other action

File: pages/settings.js
Comment (nelsonmestevao) [reply pages/settings.js]: I think it is expected to be able to correct any error or non-specific enough name. If the ninja is not mature enough for this basic responsibility, it should not even have an account and be only manage by the guardian. I guess if something unexpected happens, the organizer should block that ninja account.

**Comment by nelsonmestevao** [comment]
> I'm getting some weird behavior in the previews: when I change some info and save it, the request seems to be successful but it doesn't change it in the view until I do some other action

@miguelbrandao can you expand more on that? I don't have any problems on changing any field.

File: pages/settings.js
Comment (miguelbrandao) [reply pages/settings.js]: I really don't see why anyone should get to change their name in this case. In many platforms or services, like email and social media it makes sense, since it doesn't really matter in the grand scheme of things. But in my view we are trying to keep faithful records, and allowing for a name change without any warning to the organizers isn't really manageable. Isn't one of the reasons we have the manual account approval process so that we can approve the account and check if the information provided is accurate?

To me at least, the reasons users shouldn't be able to change their 'permanent' data here is the same reasons why UMinho's website doesn't allow you to do that.

Btw, I don't think *any* user should get to change that kind of 'permanent' data, like names and _maybe_ birthdays.

Comment (miguelbrandao) [reply pages/settings.js]: Of course there are cases where a name change would be required, like a mistake that slipped through or a legal name change, but I don't see that being a problem often enough that it would warrant allowing everyone to do it on a whim.

**Comment by miguelbrandao** [comment]
> > I'm getting some weird behavior in the previews: when I change some info and save it, the request seems to be successful but it doesn't change it in the view until I do some other action
> 
> @miguelbrandao can you expand more on that? I don't have any problems on changing any field.

Mmmmm. I was testing again to send you a screen recording but I couldn't reproduce :thinking:

**Review by miguelbrandao — APPROVED** [review]
Since I couldn't reproduce the bug, I'll approve this. Just checkout the styling errors first. The final decision on whether we allow name changes or not is yours :D 

Nice work!

File: pages/settings.js
Comment (nelsonmestevao) [reply pages/settings.js]: Well, If you decide basic information can't be changed, we just do it. And any of the cases you mention could be solved by requestion a organizer to fix it in the admin panel. I think I just followed the mock-up and it made sense to me that people could change their names as most regular platforms. I will merge like this but soon enough I should add social media fields and probably not allow name changes.

**Pull request merged**
